src/redux/books/booksThunks.js

```diff
@@ -0,0 +1,55 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+
+const URL = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/HYYvKmGO3upxaXekhp4l/books';
+const delURL = (id) => `${URL}/${id}`;
+
+export const fetchBooks = createAsyncThunk('books/fetchBooks', async () => {
+  const response = await fetch(URL)
+    .then((res) => {
+      if (!res.ok) {
+        throw new Error(`HTTP error! Status: ${res.status}`);
+      }
+      return res.json();
+    });
+  return response;
+});
+
+export const addBook = createAsyncThunk('books/addBook', async (book) => {
+  const response = await fetch(URL, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify(book),
+  })
+    .then((res) => {
+      if (!res.ok) {
+        throw new Error(`HTTP error! Status: ${res.status}`);
+      }
+      return res;
+    });
+  if (response.statusText !== 'Created') {
+    return response;
+  }
+  return book;
+});
+
+export const removeBook = createAsyncThunk('books/removeBook', async (id) => {
+  const response = await fetch(delURL(id), {
+    method: 'DELETE',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({ item_id: id }),
+  })
+    .then((res) => {
+      if (!res.ok) {
+        throw new Error(`HTTP error! Status: ${res.status}`);
+      }
+      return res;
+    });
+  if (response.statusText !== 'Created') {
+    return response;
+  }
+  return id;
```

Comment:
- [ ] As required for the project everything to still work as expected and as mentioned by the previous reviewers,  there is still an issue when a use try to `add` and `remove` a book. The action takes after refreshing the page, please 🙏🏽 correct this behavior for a better UX.